src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -784,6 +812,266 @@ private void PopulateEncLogTableParameters()
             }
         }
 
+        /// <summary>
+        /// CustomAttributes point to their target via the Parent column so we cannot simply output new rows
+        /// in the delta or we would end up with duplicates, but we also don't want to do complex logic to determine
+        /// which attributes have changes, so we just emit them all.
+        /// This means our logic for emitting CustomAttributes is to update any existing rows, either from the original
+        /// compilation or subsequent deltas, and only add more if we need to. The EncLog table is the thing that tells
+        /// the runtime which row a CustomAttributes row is (ie, new or existing)
+        /// </summary>
+        private void PopulateEncLogTableCustomAttributes()
+        {
+            if (_customAttributeParentCounts.Count == 0)
+            {
+                return;
+            }
+
+            // We need to be able to process parents in the same order as the original metadata for this algorithm to work.
+            var customAttributeParents = _customAttributeParentCounts.Keys.OrderBy(k => CodedIndex.HasCustomAttribute(k)).ToArray();
+
+            // We might need more (for deletes), but we need at least enough rows for each attribute emitted
+            var initialEncLogRowsSize = _customAttributeParentCounts.Sum(t => t.Value);
+
+            // Because we skip emitted rows when processing the original metadata, for efficiency, we can't emit log records
+            // as we go so we use a list to hold the rowIds we will need to log. We need a list because we need direct element
+            // but also need to resize because at this point we can't know how many attributes might have been deleted.
+            var encLogRows = new List<int>(Enumerable.Repeat(0, initialEncLogRowsSize));
+
+            // The first phase of this is to iterate through the (potentially huge) custom attributes from the original
+            // compilation, and issue updates for existing rows as we go.
+            var skippedParents = PopulateEncLogTableCustomAttributesFromOriginalMetadata(customAttributeParents, encLogRows);
```

Comment:
>skippedParents [](http://example.com/codeflow?start=16&length=14)

Would `addedParents` be better name - This list contains parents added since the baseline, right?